modules/m_whois.c

```diff
@@ -373,10 +375,23 @@ single_whois(struct Client *source_p, struct Client *target_p, int operspy)
 					target_p->name, buf);
 		}
 
-		sendto_one_numeric(source_p, RPL_WHOISIDLE, form_str(RPL_WHOISIDLE),
-				   target_p->name,
-				   (long)(rb_current_time() - target_p->localClient->last),
-				   (unsigned long)target_p->localClient->firsttime);
+		/* fire the doing_whois_show_idle hook to allow modules to tell us whether to show the idle time */
+		hook_data_client_approval hdata_showidle;
+
+		hdata_showidle.client = source_p;
+		hdata_showidle.target = target_p;
+		hdata_showidle.approved = 1;
+
+		call_hook(doing_whois_show_idle_hook, &hdata_showidle);
+
+		if (hdata_showidle.approved || operspy)
+		{
+			/* operspies are always allowed to see idle time */
+			sendto_one_numeric(source_p, RPL_WHOISIDLE, form_str(RPL_WHOISIDLE),
+					   target_p->name,
+					   (long)(rb_current_time() - target_p->localClient->last),
+					   (unsigned long)target_p->localClient->firsttime);
+		}
```

Comment:
Would it make sense to show the connect time even if the idle time is hidden? I don't think it gives much away.

Reply:
If we did show the connect time, we'd have to send some garbage for the idle time which could be confusing. I just decided to hide the whole thing since that's how unrealircd behaves.